pkg/reconciler/common/prune.go

```diff
@@ -35,22 +35,23 @@ import (
 )
 
 const (
-	tektonSA             = "tekton-pipelines-controller"
-	CronName             = "tekton-resource-pruner"
-	JobsTKNImageName     = "IMAGE_JOB_PRUNER_TKN"
-	pruneSkip            = "operator.tekton.dev/prune.skip"
-	pruneResources       = "operator.tekton.dev/prune.resources"
-	pruneStrategy        = "operator.tekton.dev/prune.strategy"
-	pruneKeep            = "operator.tekton.dev/prune.keep"
-	pruneKeepSince       = "operator.tekton.dev/prune.keep-since"
-	pruneSchedule        = "operator.tekton.dev/prune.schedule"
-	pruneCronLabel       = "tektonconfig.operator.tekton.dev/pruner"
-	pruneCronNsLabel     = "tektonconfig.operator.tekton.dev/pruner.ns"
-	pruneLastAppliedHash = "operator.tekton.dev/prune.hash"
-	keep                 = "keep"
-	keepSince            = "keep-since"
-	changed              = true
-	unchanged            = false
+	tektonSA               = "tekton-pipelines-controller"
+	CronName               = "tekton-resource-pruner"
+	JobsTKNImageName       = "IMAGE_JOB_PRUNER_TKN"
+	pruneSkip              = "operator.tekton.dev/prune.skip"
+	pruneResources         = "operator.tekton.dev/prune.resources"
+	pruneStrategy          = "operator.tekton.dev/prune.strategy"
+	pruneKeep              = "operator.tekton.dev/prune.keep"
+	pruneKeepSince         = "operator.tekton.dev/prune.keep-since"
+	pruneSchedule          = "operator.tekton.dev/prune.schedule"
+	pruneCronLabel         = "tektonconfig.operator.tekton.dev/pruner"
+	pruneCronNsLabel       = "tektonconfig.operator.tekton.dev/pruner.ns"
+	pruneLastAppliedHash   = "operator.tekton.dev/prune.hash"
+	nodeSelectorChangeHash = "operator.tekton.dev/nodeSelector.hash"
```

Comment:
we already have an annotation for this https://github.com/tektoncd/operator/blob/main/pkg/apis/operator/v1alpha1/const.go#L45, why to add a new one

Reply:
you mean to set the `LastAppliedHashKey` annotation on the cronJob, instead of creating a new one. 
I added a new annotaion to check if the nodeSelector changes and then to recreate the cronJob.